Lib/test/test_capi/test_opt.py

```diff
@@ -1798,6 +1798,61 @@ def testfunc(n):
         self.assertIn("_CALL_TYPE_1", uops)
         self.assertNotIn("_GUARD_IS_NOT_NONE_POP", uops)
 
+    def test_call_str_1(self):
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = str(42)
+                if y == '42':
+                    x += 1
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertIn("_CALL_STR_1", uops)
+        self.assertNotIn("_GUARD_NOS_NULL", uops)
+        self.assertNotIn("_GUARD_CALLABLE_STR_1", uops)
+
+    def test_call_str_1_result_is_str(self):
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = str(42) + 'foo'
+                if y == '42foo':
+                    x += 1
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertIn("_CALL_STR_1", uops)
+        self.assertIn("_BINARY_OP_ADD_UNICODE", uops)
+        self.assertNotIn("_GUARD_NOS_UNICODE", uops)
```

Comment:
We *should* be able to assert this too, right?

```suggestion
        self.assertNotIn("_GUARD_NOS_UNICODE", uops)
        self.assertNotIn("_GUARD_TOS_UNICODE", uops)
```

Reply:
We can, I didn't add it because TOS here is `'foo'` so not strictly what I'm testing for, but it doesn't hurt to include it. Added in [bf9ff1f](https://github.com/python/cpython/pull/132849/commits/bf9ff1f107a607b9ce4d0bb75803c101996e0d03)

Reply:
Ah I see, keeping each test separate. Either way is fine!

Reply:
I don't have a strong opinion either, but since I already added it, let's keep it :)